models/opensea/ethereum/opensea_v4_ethereum_trades.sql

```diff
@@ -27,526 +25,547 @@ with source_ethereum_transactions as (
+        end as eth_erc_idx
+    from iv_offer_consideration a
 )
-,iv_volume as (
-  select block_date
-        ,block_time
-        ,tx_hash
-        ,evt_index
-        ,max(token_contract_address) as token_contract_address
-        ,CAST(sum(case when is_price then original_amount end) AS DECIMAL(38,0)) as price_amount_raw
-        ,sum(case when is_platform_fee then original_amount end) as platform_fee_amount_raw
-        ,max(case when is_platform_fee then receiver end) as platform_fee_receiver
-        ,sum(case when is_creator_fee then original_amount end) as creator_fee_amount_raw
-        ,sum(case when is_creator_fee and creator_fee_idx = 1 then original_amount end) as creator_fee_amount_raw_1
-        ,sum(case when is_creator_fee and creator_fee_idx = 2 then original_amount end) as creator_fee_amount_raw_2
-        ,sum(case when is_creator_fee and creator_fee_idx = 3 then original_amount end) as creator_fee_amount_raw_3
-        ,sum(case when is_creator_fee and creator_fee_idx = 4 then original_amount end) as creator_fee_amount_raw_4
-        ,sum(case when is_creator_fee and creator_fee_idx = 5 then original_amount end) as creator_fee_amount_raw_5
-        ,max(case when is_creator_fee and creator_fee_idx = 1 then receiver end) as creator_fee_receiver_1
-        ,max(case when is_creator_fee and creator_fee_idx = 2 then receiver end) as creator_fee_receiver_2
-        ,max(case when is_creator_fee and creator_fee_idx = 3 then receiver end) as creator_fee_receiver_3
-        ,max(case when is_creator_fee and creator_fee_idx = 4 then receiver end) as creator_fee_receiver_4
-        ,max(case when is_creator_fee and creator_fee_idx = 5 then receiver end) as creator_fee_receiver_5
-        ,max(a.fee_wallet_name) as fee_wallet_name
-
-   from iv_base_pairs a
-  where 1=1
-    and eth_erc_idx > 0
-  group by 1,2,3,4
-  having count(distinct token_contract_address) = 1  -- some private sale trade has more that one currencies
+
+, iv_base_pairs as (
+    select 
+          a.block_time
+        , a.block_number
+        , a.tx_hash
+        , coalesce(b.om_evt_index, a.evt_index) as evt_index  -- when order_matched exists, then replace evt_index to its
+        , a.sub_type
+        , a.sub_idx
+        , a.offer_first_item_type
+        , a.consideration_first_item_type
+        , a.offerer
+        , a.recipient
+        , a.sender
+        , a.receiver
+        , a.zone
+        , a.token_contract_address
+        , a.original_amount
+        , a.item_type
+        , a.token_id
+        , a.platform_contract_address
+        , a.offer_cnt
+        , a.consideration_cnt
+        , a.order_hash
+        , a.is_private
+        , b.om_evt_index
+        , b.om_order_id
+
+        , case 
+            when sender = receiver then true 
+            else false 
+        end as is_self_trans
+
+        , case 
+            when f.wallet_address is not null then true 
+            else false 
+        end as is_platform_fee
```

Comment:
Hi @DevCav, since `f` is defined in line 121, the wallet_address may not be null.
Maybe you mean the result of `a left join f`.
It seems that this spell is too complicated, and logic is a bit messy,
do you think you can break down the sql and examine the query logic again?